cmd/handler-utils.go

```diff
@@ -358,6 +359,31 @@ func httpTraceHdrs(f http.HandlerFunc) http.HandlerFunc {
 	}
 }
 
+// maxClients throttles the S3 API calls
+func maxClients(f http.HandlerFunc, enabled bool, connMaxCh chan struct{}, connDeadline time.Duration) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if !enabled {
+			f.ServeHTTP(w, r)
+			return
+		}
+
+		select {
+		case connMaxCh <- struct{}{}:
+			f.ServeHTTP(w, r)
+			<-connMaxCh
```

Comment:
defer this, so if a panic happens we always release this.

```suggestion
			defer func() {<-connMaxCh}()
			f.ServeHTTP(w, r)
```